dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ElasticManager.java

```diff
@@ -92,7 +92,6 @@ private String getDocumentValueFromEntry(Entry entry) {
      * @param command The command to perform for the document, either "update" or "delete" document
      */
     public void handleIndexUpdate(Entry entry, ElasticMode command) {
-        entry = filterCheckerWorkflows(entry);
```

Comment:
Missed this on the first review, why is the filter for checker workflows being removed actually?

Reply:
Removal of this also allows a tool's checker workflow to leak through to the index.

Reply:
As I understand it, we want checker workflows to be present in the index when the parent is published and absent if the parent is unpublished. If there is a checker attached to a workflow when it is published, publish both the checker and parent workflow. 

I believe in order to be able to search checkers properly, you want to allow the ES index to be given updates about them. Perhaps this removal is an opinionated approach to try to decouple the ES manager and business logic from each other. If it's intended that checker workflows can't have their ES documents updated, do we expect them to be orphaned and then added again if their properties change? @garyluu can you explain the failure mode a little more?

Reply:
Oh crap, I see. 
I think there was a misunderstanding. 
Checker workflows used to be in search but we removed them totally with https://github.com/ga4gh/dockstore/pull/2006

The issue isn't so much that checker workflows are improperly being removed, but we thought that Louise was reporting that they were somehow being leaking into the search despite the filter.

Reply:
Ah, my bad, that helps! Is it possible this is actually fixed in develop, just that the state of the database or index on production is off?

Reply:
very possible